### DISCUSSION TITLE
为Qexo的友链页面贡献一套修改过的样式，借此为Qexo的发展尽我的绵薄之力

### DISCUSSION THREAD

**Discussion by Tseshongfeeshur** [Qexo/Qexo#459] [Ideas]
# 文件

[点击下载](https://github.com/Qexo/Qexo/files/12774436/LINK.zip)

（html部分和官网相同，只不过更换了css和js引用，非常随和:smile:）

# 简介

我现在是一名高中生，一直在用Qexo来管理博客。Qexo是一个高效且易用的工具，管理界面十分美观。可是，自带的友链样式却稍显草率。我已经发现的问题包括以下几点（嘴笨不会说话，请谅解）：

1. 对于加载动画，文字和不太透明的背景显得多余；
2. 友情链接分块没有圆角；
3. 配色有些平淡；
4. 阴影过于厚重；
5. 友情链接一行两个有点浪费空间，不太适合查看；
6. 图片加载失败后占位符号不太美观；
7. 头像旋转的特效并不需要，反而显得花哨；
8. 在链接描述为空时，排版会出现错乱的问题；
9. 文字部分整体轻重不明显；
10. 对于超链接，只有点击文字才能跳转，而鼠标悬停在块上时已经提前出现视觉反馈，不符合大多数用户操作逻辑；
11. 使用字体比较死板。

虽然我并没有用到关于友情链接的管理，但为了完善Qexo的外观，同时为同学的网站做一些观感优化，我修改了全套样式文件。

# 效果

![Sst](https://github.com/Qexo/Qexo/assets/99335871/f8617bbc-94f7-4e8f-8c02-365034cd3592)

# 所做优化

1. 删除加载动画下面的文字，删除背景；
2. 增加圆角；
3. 将主题色更换为蓝色；
4. 减淡阴影；
5. 友情链接一行分布四个；
6. 添加图片加载失败后的替换图片；
7. 删除头像旋转特效，并增加鼠标悬浮时的淡入淡出动画；
8. 在链接描述为空时，将描述改为标题；
9. 删除粗体，调整字号和颜色；
10. 将超链接标签应用在整个块元素上；
11. 使用Bilibili同款字体。

# 关于作者

[个人博客](https://blog.xn--tiyy81g.eu.org/)

~~不久后可能还会更改说说的样式~~

**Comment by am-abudu** [comment]
十分感谢你的支持！你可以直接于 Qexo/Qstatic 提交Pr!

**Comment by Fgaoxing** [comment]
@Tseshongfeeshur 非常不错，不过请到友链仓库进行提交，需要注意的是，Qexo是对各个主题做适配，如果你是对某个主题做适配，可以提交到对应的文件夹，然后提交文档即可，我会帮你整理好

**Reply by Tseshongfeeshur** [reply]
嗯，还不算针对主题适配，因为目标主题拥有自己的友链管理模式。这个优化只能算是对于默认样式外观的调整，也只是从外观上适配了目标主题。我能提交到 `Qexo/Qstatic/hexo/friends_ryan.css` 和 `Qexo/Qstatic/hexo/friends_ryan.js` 吗？

**Comment by Fgaoxing** [comment]
非常不错，不过请到友链仓库进行提交，需要注意的是，Qexo是对各个主题做适配，如果你是对某个主题做适配，可以提交到对应的文件夹，然后提交文档即可，我会帮你整理好

**Reply by Tseshongfeeshur** [reply]
嗯，还不算针对主题适配，因为目标主题拥有自己的友链管理模式。这个优化只能算是对于默认样式外观的调整，也只是从外观上适配了目标主题。我能提交到 `Qexo/Qstatic/hexo/friends_ryan.css` 和 `Qexo/Qstatic/hexo/friends_ryan.js` 吗？

**Reply by am-abudu** [reply]
可以的。非常感谢您的贡献

**Comment by am-abudu** [comment]
我已经完成了目录结构的重整以及你的样式的加入，现在只差文档的更新了

**Reply by Tseshongfeeshur** [reply]
好的，感谢您 :smile:
更新文档需要我提交Pr吗？如果需要标明作者的话，标注 `Ryan璃黯` 就行 :wink:
那之后如果样式有更新，还是修改Qstatic的对应文件后提交Pr就行吗

**Reply by am-abudu** [reply]
是的。我暂时没有空闲去更新文档，如果你能提交是最好的。

**Reply by Tseshongfeeshur** [reply]
今天我看了一眼文档，发现友链文档的排版不是很直观，能不能改一下目录层级之间的关系？
比如现在的目录结构是：
> ## 主题适配
> ## 友链申请主题适配

可以改成：
> ## 外观美化
> ### 主题适配
> #### 友链展示主题适配
> ##### Yun
> ...
> ##### Jian
> ...
> #### 友链申请主题适配
> ##### Butterfly
> ...
> ### 非主题适配
> #### 友链展示外观美化
> ##### **圆角风格**
> **作者：Ryan璃黯**

这样，目录层级可能会更清楚。而且Butterfly的那一大段建议使用引用。

~~（我是不是有点过分了）~~

**Reply by Tseshongfeeshur** [reply]
目前我就只添加我的部分吧，其他的地方不改了。
